enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+"image" resource that is embedded in the "cluster" resource, but for
+kubernetes-native it is more natural for it to be separate.
+
+The discovery ISO can be downloaded from a URL that is available in the
+resource's Status.
+
+The details of this resource definition are being discussed [in a
+pull request](https://github.com/openshift/assisted-service/pull/969).
+
+**ClusterDeployment**
+Hive's ClusterDeployment CRD will be extended to include all cluster details
+that the agent-based install needs. There will not be a new cluster resource.
+The contents of this API correlate to the "cluster" resource in the assisted
+installer's current REST API.
+
+The details of this are being discussed on a Hive
+[pull request](https://github.com/openshift/hive/pull/1247).
+
+
+**Agent**
+Agent is a new resource, part of the assisted installer's new operator, that
+represents a host that is destined to become part of an OpenShift cluster, and
+is running an agent that is able to run inspection and installation tasks. It
+correlates to the "host" resource in the Assisted Installer's REST API.
+
+The details of this API are being discussed in a [pull
+request](https://github.com/openshift/assisted-service/pull/861) that
+implements the CRD.
+
+
+#### REST API Access
+
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download ISO and PXE artifacts: These files must be available for download
+via HTTP, either directly by users or by a BMC.  Because BMCs do not pass
+authentication headers, the Assisted Service must generate some URL or query
+parameter so that the ISO’s location isn’t easily guessable.
+* Agent APIs (near-term): Until a point where the agent creates and modifies
+Agent CRs itself, the agent will continue communicating with the service via
+REST APIs. Currently the service embeds the user’s pull secret in the discovery
+ISO which the agent passes in an authentication header.  In this case the
+service can generate and embed some token which it can later validate.
+
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include a new `AgentBareMetal` platform that
+contains such fields as:
+
+* API VIP
+* API VIP DNS name
+* IngressVIP
+
+A new `InstallStrategy` section of the Spec enables the API to describe a way
+of installing a cluster other than the default of using `openshift-install`.
+The new field has an "agent" option that would be common to any Agent-driven
+installation, regardless of platform. Fields include:
+
+* AgentSelector, a label selector for identifying which Agent resources should
+be included in the cluster.
+* ProvisionRequirements, where the API user can specify how many agents to
```

Comment:
You know, somewhere earlier in this doc, we should mention that the initial focus for supporting cluster provisioning is single-node bootstrap-in-place

Reply:
I agree that it is the initial focus and should be mentioned, but this enhancement covers both single-node and highly-available control plane clusters.